src/main/java/com/scaleunlimited/flinkcrawler/tools/CrawlTopology.java

```diff
@@ -165,7 +165,7 @@ public void stop() throws Exception {
         private BaseHttpFetcherBuilder _robotsFetcherBuilder = new SimpleHttpFetcherBuilder(INVALID_USER_AGENT);
         private SimpleRobotRulesParser _robotsParser = new SimpleRobotRulesParser();
 
-        private BaseUrlLengthener _urlLengthener = new SimpleUrlLengthener();
+        private BaseUrlLengthener _urlLengthener = new SimpleUrlLengthener(new SimpleHttpFetcherBuilder(INVALID_USER_AGENT));
```

Comment:
How does the user agent get changed if the user specifies it?

Reply:
I think I'm going to wait for @vmagotra 's changes to support specifying the user agent on the command line before we address this issue for the URL lengthener. If I get this pull request complete today, then maybe he can just handle this code along with the main fetcher, robts fetcher, etc.